app/src/test/java/com/farmerbb/taskbar/ui/TaskbarControllerTest.java

```diff
@@ -0,0 +1,220 @@
+        checkStartButtonPadding(padding, startButton);
+
+        prefs.edit().putString(PREF_START_BUTTON_IMAGE, "non-support").apply();
+        callDrawStartButton(context, startButton, prefs);
+        checkStartButtonPadding(0, startButton);
+    }
+
+    @Test
+    public void testGetTaskbarGravity() {
+        assertEquals(
+                Gravity.BOTTOM | Gravity.LEFT,
+                callGetTaskbarGravity(POSITION_BOTTOM_LEFT)
+        );
+        assertEquals(
+                Gravity.BOTTOM | Gravity.LEFT,
+                callGetTaskbarGravity(POSITION_BOTTOM_VERTICAL_LEFT)
+        );
+        assertEquals(
+                Gravity.BOTTOM | Gravity.RIGHT,
+                callGetTaskbarGravity(POSITION_BOTTOM_RIGHT)
+        );
+        assertEquals(
+                Gravity.BOTTOM | Gravity.RIGHT,
+                callGetTaskbarGravity(POSITION_BOTTOM_VERTICAL_RIGHT)
+        );
+        assertEquals(
+                Gravity.TOP | Gravity.LEFT,
+                callGetTaskbarGravity(POSITION_TOP_LEFT)
+        );
+        assertEquals(
+                Gravity.TOP | Gravity.LEFT,
+                callGetTaskbarGravity(POSITION_TOP_VERTICAL_LEFT)
+        );
+        assertEquals(
+                Gravity.TOP | Gravity.RIGHT,
+                callGetTaskbarGravity(POSITION_TOP_RIGHT)
+        );
+        assertEquals(
+                Gravity.TOP | Gravity.RIGHT,
+                callGetTaskbarGravity(POSITION_TOP_VERTICAL_RIGHT)
+        );
+        assertEquals(
+                Gravity.BOTTOM | Gravity.LEFT,
+                callGetTaskbarGravity("unsupported")
+        );
+    }
+
+    @Test
+    public void testGetTaskbarLayoutId() {
+        assertEquals(
+                R.layout.tb_taskbar_left,
+                callGetTaskbarLayoutId(POSITION_BOTTOM_LEFT)
+        );
+        assertEquals(
+                R.layout.tb_taskbar_vertical,
+                callGetTaskbarLayoutId(POSITION_BOTTOM_VERTICAL_LEFT)
+        );
+        assertEquals(
+                R.layout.tb_taskbar_right,
+                callGetTaskbarLayoutId(POSITION_BOTTOM_RIGHT)
+        );
+        assertEquals(
+                R.layout.tb_taskbar_vertical,
+                callGetTaskbarLayoutId(POSITION_BOTTOM_VERTICAL_RIGHT)
+        );
+        assertEquals(
+                R.layout.tb_taskbar_left,
+                callGetTaskbarLayoutId(POSITION_TOP_LEFT)
+        );
+        assertEquals(
+                R.layout.tb_taskbar_top_vertical,
+                callGetTaskbarLayoutId(POSITION_TOP_VERTICAL_LEFT)
+        );
+        assertEquals(
+                R.layout.tb_taskbar_right,
+                callGetTaskbarLayoutId(POSITION_TOP_RIGHT)
+        );
+        assertEquals(
+                R.layout.tb_taskbar_top_vertical,
+                callGetTaskbarLayoutId(POSITION_TOP_VERTICAL_RIGHT)
+        );
+        assertEquals(
+                R.layout.tb_taskbar_left,
+                callGetTaskbarLayoutId("unsupported")
+        );
+    }
+
```

Comment:
One more question, is there any advantage to using Robolectric's `callInstanceMethod()` over just calling the method directly?  You can make the method public and then annotate it with `@VisibleForTesting`.  This avoids the need for reflection and cleans up the test suite a bit.